### PR TITLE
mlx5 5.0 patch

### PR DESCRIPTION
**Pull request by kierank**
~- I need to remove txsync from the header and put it in the second header and have it build. They have separated tx and rx code so you can't just include a header with both functions any more~

### CODE REVIEW

**Comment by kierank** [comment]
@andrewbonney @stuartgrace-bbc Would it be possible for you to review this patch?

File: LINUX/final-patches/mellanox--mlx5--5.0 [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
++
+ 	if (MLX5E_GET_PFLAG(&priv->channels.params, MLX5E_PFLAG_SNIFFER)) {
+ 		mlx5e_sniffer_stop(priv);
+ 		MLX5E_SET_PFLAG(&priv->channels.params, MLX5E_PFLAG_SNIFFER, 0);
+@@ -3542,6 +3590,10 @@ int mlx5e_close_locked(struct net_device *netdev)
+ 	mlx5e_deactivate_priv_channels(priv);
+ 	mlx5e_close_channels(&priv->channels);
+ 
++#ifdef DEV_NETMAP
++	netmap_enable_all_rings(netdev); /* NOP if netmap not in use */
```
Comment (vrangayyan) [inline LINUX/final-patches/mellanox--mlx5--5.0:252]: This seems to be in an odd place. It should be in mlx5e_open or mlx5e_open_locked no?

Comment (kierank) [reply LINUX/final-patches/mellanox--mlx5--5.0:252]: You are indeed correct, not sure how this happened exactly but managing diffs of diffs is headache inducing.
Will fix tomorrow

Comment (kierank) [reply LINUX/final-patches/mellanox--mlx5--5.0]: Thanks, should be resolved now. I'm happy to address any other issues you might find.

**Comment by kierank** [comment]
Seems to work for me so I remove the Work in Progress label

**Comment by kierank** [comment]
Thanks, I will be submitting a patch for mlx5 5.1 in the near future

**Comment by giuseppelettieri** [comment]
Thanks to you, looking forward to it.

**Pull request merged**